### ISSUE TITLE
Open big files seems too slow

### ISSUE THREAD

**Issue by maplewizard** [syl20bnr/spacemacs#3491]
I opened a file(4.6 M) in emacs on Mac OS X 10.11, but it seems too slow for me. I tried to stop all the minor mode but it seems of no help. Can any one give a solution?

**Comment by TheBB** [comment]
Is it still slow if you open it literally (`SPC f l`)?

**Comment by maplewizard** [comment]
It helps a lot, but what does that mean?

**Comment by zilongshanren** [comment]
@maplewizard  Try to open this file with a fresh Emacs or Sublime Text and compare the result.
Note: You will find Sublime text is also very slow when opening big files.

**Comment by d12frosted** [comment]
Once I needed to open and edit 146MB js file. Don't laugh, `emacs -Q` was my friend :D

**Comment by TheBB** [comment]
It doesn't enable any syntax highlighting or minor modes. Try turning of `show-smartparens-global-mode` for example. It tends to slow things down.

**Comment by syl20bnr** [comment]
@d12frosted You should be able to open it literally with `SPC f l`. If the performance are abnormally slow then open a bug, Spacemacs should allow to open big files. I know we have fixed an issue with `evil-search-highlight-persist` which was activated in fundamental mode, we can fix others as we discover them, this is important.

**Comment by ralesi** [comment]
I have a hook in my config to automatically open literally with large files over a certain size.  Do we want this option?

**Comment by d12frosted** [comment]
@syl20bnr and @TheBB  thanks for the tip about `SPC f l`! Just tested on HUUGE file - works nicely. 

@ralesi will not hurt sharing it (at least)! :smile:

**Comment by TheBB** [comment]
@ralesi If it can do a prompt instead of just deciding automatically, I'm all for it.

**Comment by syl20bnr** [comment]
@ralesi and @TheBB :+1:

**Comment by mukhali** [comment]
maybe need http://melpa.org/#/vlf ?

**Comment by ralesi** [comment]
How about this? 

````
  ;; check when opening large files
  (defun spacemacs/check-large-file ()
    (when (> (buffer-size) (* 1024 1024))
      (when (y-or-n-p "Open file literally?")
        (setq buffer-read-only t)
        (buffer-disable-undo)
        (fundamental-mode))))

  (add-hook 'find-file-hook 'spacemacs/check-large-file)
````

**Comment by TheBB** [comment]
@ralesi That seems to work fine here. Maybe the prompt can include some more info, e.g. "This is a large file, open literally to avoid performance problems?"

If you make a PR I'll merge it.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
